### PR TITLE
Refingng the update project controller

### PR DESCRIPTION
**Pull request by C0SMlC**
Added more fields for updates.

### CODE REVIEW

**Comment by Shivz3232** [comment]
Please update the sanitizer for this controller accordingly

File: src/controllers/project.controller/updateProject/index.js [hunk 1]
```diff
@@ -19,10 +21,27 @@ const updateProject = catchAsync(async (req, res) => {
...
     changes: {},
   };
-  // if (req.body.caption) variables['changes']['caption'] = req.body.caption;
+
   if (req.body.description) variables.changes.description = req.body.description;
   if (req.body.title) variables.changes.title = req.body.title;
   if (req.body.link) variables.changes.link = req.body.link;
+  if (req.body.status !== undefined) variables.changes.status = req.body.status;
+  if (req.body.completed !== undefined) variables.changes.completed = req.body.completed;
+  if (req.body.looking_for_members !== undefined) variables.changes.looking_for_members = req.body.looking_for_members;
```
Comment (Shivz3232) [inline src/controllers/project.controller/updateProject/index.js]: Prameters from here on do not belong to the project table, you will have to update them on their separate tables like team, team_role_requirements etc

Comment (C0SMlC) [reply src/controllers/project.controller/updateProject/index.js]: Done

File: src/controllers/project.controller/updateProject/index.js [hunk 2]
```diff
@@ -19,13 +21,81 @@ const updateProject = catchAsync(async (req, res) => {
...
+  if (req.body.project_tags) variables.changes.project_tags = req.body.project_tags;
+  if (req.body.documents) variables.changes.documents = req.body.documents;
+
+  if (req.body.looking_for_members || req.body.looking_for_mentors) {
+    const teamName = req.body.team_name ? req.body.team_name : `${req.body.title} team`;
+    const teamOnInovact = req.body.team_on_inovact;
+    const teamCreated = await createDefaultTeam(id, teamName, req.body.looking_for_mentors, req.body.looking_for_members, teamOnInovact);
+    variables.changes.team_id = teamCreated.team_id;
+  } else if (req.body.looking_for_members === false && req.body.looking_for_mentors === false) {
+    variables.changes.team_id = null;
```
Comment (Shivz3232) [inline src/controllers/project.controller/updateProject/index.js:60]: Inside this else block along with setting team_id null, we will have to delete that team from the teams table.

File: src/controllers/project.controller/updateProject/index.js [hunk 3]
```diff
@@ -19,13 +21,81 @@ const updateProject = catchAsync(async (req, res) => {
...
+  if (req.body.looking_for_mentors !== undefined || req.body.looking_for_members !== undefined) {
+    const projectFlagsUpdateVariables = {
+      projectId: id,
+      lookingForMentors: req.body.looking_for_mentors,
+      lookingForMembers: req.body.looking_for_members,
+    };
+    await Hasura(updateProjectFlags, projectFlagsUpdateVariables);
+  }
+
+  if (req.body.mentions) variables.changes.mentions = req.body.mentions;
```
Comment (Shivz3232) [inline src/controllers/project.controller/updateProject/index.js]: I see you are making use of the same `variables.changes` here, but I think you missed using it in a query...

Comment (Shivz3232) [reply src/controllers/project.controller/updateProject/index.js]: Oh, you are updating mentions, tags and documents separately, so you can remove L53, 54 and 55

File: src/controllers/project.controller/updateProject/index.js [hunk 4]
```diff
@@ -19,13 +21,81 @@ const updateProject = catchAsync(async (req, res) => {
...
+          name: tag_name.toLowerCase(),
+        },
+        on_conflict: {
+          constraint: 'hashtag_tag_name_key',
+          update_columns: 'name',
+        },
+      },
+      project_id: id,
+    }));
+    await Hasura(updateProjectTags, { objects: tags });
```
Comment (Shivz3232) [inline src/controllers/project.controller/updateProject/index.js:85]: Delete older tags.

File: src/controllers/project.controller/updateProject/index.js [hunk 5]
```diff
@@ -19,13 +21,81 @@ const updateProject = catchAsync(async (req, res) => {
...
+  }
+
+  if (req.body.mentions && req.body.mentions.length > 0) {
+    const mentions = req.body.mentions.map((user_id) => ({
+      user_id,
+      project_id: id,
+    }));
+    await Hasura(updateMentions, { objects: mentions });
```
Comment (Shivz3232) [inline src/controllers/project.controller/updateProject/index.js:93]: Same here, delete existing mentions and then save the new ones

Comment (C0SMlC) [reply src/controllers/project.controller/updateProject/index.js:93]: already there

File: src/controllers/project.controller/updateProject/index.js [hunk 6]
```diff
@@ -1,7 +1,8 @@
 const { validationResult } = require('express-validator');
 const catchAsync = require('../../../utils/catchAsync');
 const { query: Hasura } = require('../../../utils/hasura');
-const { updatePost_query } = require('./queries/queries');
+const { updatePost_query, updateRolesRequired, updateProjectFlags, updateDocuments, updateProjectTags, updateMentions, deleteTeam } = require('./queries/mutations');
```
Comment (Shivz3232) [inline src/controllers/project.controller/updateProject/index.js]: Please use strict camel case for all variable names updatePostQuery or updatePostMutation

File: src/controllers/project.controller/updateProject/queries/mutations.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+const deleteTeam = `mutation deleteTeam($projectId: Int) {
+  delete_team(where: { id: { _eq: $projectId } }) {
+    affected_rows
+  }
+}
+`;
+
+module.exports = {
+  updatePost_query,
```
Comment (Shivz3232) [inline src/controllers/project.controller/updateProject/queries/mutations.js]: .

**Review by Shivz3232 — APPROVED** [review]

**Pull request merged**